lib/types.nix

```
@@ -443,6 +443,10 @@ rec {
       descriptionClass = "noun";
       check = x: str.check x && builtins.match pattern x != null;
       inherit (str) merge;
+      functor = defaultFunctor "strMatching" // {
+        payload = pattern;
```

Quick thing for forward compatibility:

```suggestion
        payload = { inherit pattern; };
```